### PR TITLE
feat: Upgrade to Keptn 0.17

### PR DESCRIPTION
**Pull request by TannerGabriel**
## This PR

- Upgrades integration tests to support Keptn v0.16 and v0.17
- Document the installation for Keptn v0.17

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -70,16 +70,24 @@ jobs:
...
           KEPTN_VERSION: ${{ matrix.keptn-version }}
-          KEPTN_INSTALL_PARAMETERS: -v --endpoint-service-type=LoadBalancer
+          HELM_VALUES: |
+            control-plane:
+              apiGatewayNginx:
+                type: LoadBalancer
+              features:
+                automaticProvisioningURL: http://keptn-gitea-provisioner-service.default # Keptn 0.16 compatability 
+                automaticProvisioning:
+                  serviceURL: http://keptn-gitea-provisioner-service.default
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml]: Please note: You have to move the keptn isntall step AFTER you installed gitea provisioner service

File: README.md [hunk 2]
```diff
@@ -23,16 +23,15 @@ instance:
   VERSION=0.1.0
   NAMESPACE=default
   GITEA_ENDPOINT="http://gitea-http.${NAMESPACE}:3000/"
-   GITEA_ADMIN_USERNAME=#Define a username for the admin
-   GITEA_ADMIN_PASSWORD=#Define a password for the admin
+  GITEA_ADMIN_USERNAME=#Define a username for the admin
+  GITEA_ADMIN_PASSWORD=#Define a password for the admin
```
Comment (Raffy23) [inline README.md]: Please keep at lest the password intended as a leading whitespace will not save the command in the history of bash

File: README.md [hunk 3]
```diff
@@ -57,11 +56,16 @@ instance:
       --set gitea.config.server.ROOT_URL=http://gitea-http.${NAMESPACE}:3000/
   ```
 
-* Keptn must be configured to use the keptn-gitea-provisioner-service to automatically provision git repositories:
+* Keptn must be configured to use the keptn-gitea-provisioner-service to automatically provision git repositories. The flag is different for Keptn version 0.16 and version 0.17 and onwards:
   ```bash
   #!/bin/bash
   NAMESPACE=default
   
+  # Keptn 0.17
```
Comment (christian-kreuzberger-dtx) [inline README.md:64]: I think we should use heaidng 3, e.g., 
```suggestion
### Keptn 0.17
```
here

File: .github/workflows/integration_tests.yml [hunk 4]
```diff
@@ -70,16 +70,24 @@ jobs:
...
       - name: Install Keptn
         id: install_keptn
-        uses: keptn-sandbox/action-install-keptn@v1.0.0
+        uses: keptn-sandbox/action-install-keptn@v2.0.0
         timeout-minutes: 10
         with:
           KEPTN_VERSION: ${{ matrix.keptn-version }}
-          KEPTN_INSTALL_PARAMETERS: -v --endpoint-service-type=LoadBalancer
+          HELM_VALUES: |
+            control-plane:
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml]: the field `control-plane:` has also been removed, so the correct HELM values here for 0.17 and beyond would be
```
            apiGatewayNginx:
              type: LoadBalancer
            features:
              automaticProvisioning:
                serviceURL: http://keptn-gitea-provisioner-service.default
```

File: README.md [hunk 5]
```diff
@@ -57,11 +56,16 @@ instance:
...
+  helm upgrade -n keptn keptn keptn/keptn \
+    --set "control-plane.features.automaticProvisioning.serviceURL=http://keptn-gitea-provisioner-service.${NAMESPACE}"
```
Comment (christian-kreuzberger-dtx) [inline README.md]: You have to remove `control-plane.` from this

Comment (TannerGabriel) [reply README.md]: Has been removed!

File: README.md (hunk 2 above)
Comment (TannerGabriel) [reply README.md]: Oh, I didn't know that. Added the whitespaces again.

**Comment by TannerGabriel** [comment]
Current PR run: https://github.com/keptn-sandbox/keptn-gitea-provisioner-service/actions/runs/2647515672

File: README.md [hunk 6]
```diff
@@ -22,8 +23,8 @@ instance:
   VERSION=0.1.0
   NAMESPACE=default
   GITEA_ENDPOINT="http://gitea-http.${NAMESPACE}:3000/"
-  GITEA_ADMIN_USERNAME=#Define a username for the admin
```
Comment (Raffy23) [inline README.md]: Is there any special reason that the `GITEA_ENDPOINT` variable was removed? I would prefer the definition of the endpoint up here and use it in the helm install later on

**Review by Raffy23 — APPROVED** [review]
LGTM

**Pull request merged**